ocaml/xapi/vgpuops.ml

```diff
@@ -129,7 +129,8 @@ let reserve_free_virtual_function ~__context vm pf =
 let add_vgpus_to_vm ~__context host vm vgpus vgpu_manual_setup =
   (* Only support a maximum of one virtual GPU per VM for now. *)
   match vgpus with
-  | [] -> ()
+  | [] ->
+    (try Db.VM.remove_from_other_config ~__context ~self:vm ~key:Xapi_globs.vgpu_pci with _ -> ());
```

Comment:
Oh, I didn't look carefully: I think it should be outside of the match, so that it is clear that the key is _always_ reset. That is how it was in the past. You don't need to worry about the race condition you are taking about, because this function is called with a global lock held from message_forwarding.